### PR TITLE
[Utils] Expand docker images, add check-all

### PR DESCRIPTION
**Pull request by dmbryson**
Expands docker-utils image support to all three Ubuntu versions on which
Swift is built. Adds a check-all script that uses docker to build and
test llbuild in all supported images.

### CODE REVIEW

**Comment by dmbryson** [comment]
Ubuntu 14.04 needs to be tweaked a bit before merge.  As it currently stands, it fails due to the clang version in it not supporting c++14.

File: llbuild.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1407,6 +1413,18 @@
 			path = Windows;
 			sourceTree = "<group>";
 		};
+		406A04FC21627FD900EBA895 /* docker */ = {
+			isa = PBXGroup;
+			children = (
```
Comment (jakepetroules) [inline llbuild.xcodeproj/project.pbxproj]: 👍 :)

File: utils/build-and-test [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+if [ "$#" -ne 1 ]; then
+    echo "$0: error: usage: $0 <target-platform>"
+    exit 1
+fi
+
+TARGET=$1
+SRCROOT=${PWD}
+BUILDROOT=.build/.ci/${TARGET}
+
+mkdir -p ${BUILDROOT}
```
Comment (jakepetroules) [inline utils/build-and-test]: Double-quote ${BUILDROOT}

File: utils/build-and-test [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+cd ${BUILDROOT}
```
Comment (jakepetroules) [inline utils/build-and-test]: Also here

File: utils/build-and-test [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+cmake -G Ninja -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ ${SRCROOT}
```
Comment (jakepetroules) [inline utils/build-and-test]: Double quote ${SRCROOT}

File: utils/check-all [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+for c in pip docker; do
+  command -v $c >/dev/null 2>&1 || { echo >&2 "error: $c required, but not found"; exit 1; }
+done
+
+# Ensure the docker dependencies are installed
+pip install -r utils/docker/requirements.txt
+
+# Check all targets
+for TARGET in ${TARGETS}; do
+    utils/docker/docker-utils build --target=${TARGET}
```
Comment (jakepetroules) [inline utils/check-all]: Double quote ${TARGET}

File: utils/check-all [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    utils/docker/docker-utils run --target=${TARGET} utils/build-and-test ubuntu-${TARGET}
```
Comment (jakepetroules) [inline utils/check-all]: Double quote both arguments containing ${TARGET}

File: utils/docker/Dockerfile-14.04 [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    sqlite3 \
+    python-pip \
+    libsqlite3-dev \
+    libncurses5-dev
+RUN pip install lit
+RUN apt-get -q install -y llvm-3.6-tools
+RUN update-alternatives --install /bin/sh sh /bin/bash 100
+
+ARG SNAPSHOT
+COPY $SNAPSHOT /
```
Comment (jakepetroules) [inline utils/docker/Dockerfile-14.04]: Double quote $SNAPSHOT

File: utils/docker/Dockerfile-14.04 [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+RUN tar -xvzf $SNAPSHOT --directory / --strip-components=1 && \
```
Comment (jakepetroules) [inline utils/docker/Dockerfile-14.04]: Double quote $SNAPSHOT

File: utils/docker/Dockerfile-18.04 [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    sqlite3 \
+    python-pip \
+    libsqlite3-dev \
+    libncurses5-dev
+RUN pip install lit
+RUN apt-get -q install -y llvm-6.0-tools
+RUN update-alternatives --install /bin/sh sh /bin/bash 100
+
+ARG SNAPSHOT
+COPY $SNAPSHOT /
```
Comment (jakepetroules) [inline utils/docker/Dockerfile-18.04]: Double quote $SNAPSHOT

File: utils/docker/Dockerfile-18.04 [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+RUN tar -xvzf $SNAPSHOT --directory / --strip-components=1 && \
```
Comment (jakepetroules) [inline utils/docker/Dockerfile-18.04]: Double quote $SNAPSHOT

File: utils/docker/Dockerfile-16.04 [hunk 11]
```diff
@@ -30,4 +30,4 @@ RUN tar -xvzf $SNAPSHOT --directory / --strip-components=1 && \
 # Set Swift Path
```
Comment (jakepetroules) [inline utils/docker/Dockerfile-16.04]: Double quote both occurrences of $SNAPSHOT in this file to match

**Comment by dmbryson** [comment]
Ah yes, spaces in paths are the worst.  ;-)

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Pull request merged**